mqtt/mqttd/src/app/mod.rs

```diff
@@ -107,7 +112,13 @@ pub trait Bootstrap {
     ) -> Result<(Broker<Self::Authorizer>, FilePersistor<VersionedFileFormat>)>;
 
     /// Returns update interval for snapshotter.
-    fn snapshot_interval(&self, settings: &Self::Settings) -> std::time::Duration;
+    fn snapshot_interval(&self, settings: &Self::Settings) -> Duration;
+
+    /// Returns session expiration.
+    fn session_expiration(&self, settings: &Self::Settings) -> Duration;
+
+    /// Returns session cleanup interval.
+    fn session_cleanup_interval(&self, settings: &Self::Settings) -> Duration;
```

Comment:
nit: If these can be obtained from the Settings do we need them in bootstrap? 

The pattern we have now is to pass Settings into bootstrap, then don't touch settings again. But this seems to cause some duplicated code. Maybe we should be extracting from Settings directly for simple cases?

Reply:
Yeah, it seems weird, but here is what I understand. We have an `App<B>` struct that is a generic on `Bootstrap` which has associated type `Bootstrap::Settings`. So `App` does not know the type of Settings to extract any values from it, so `Bootstrap` provides "getters" via its interface. 

As an alternative, we can intro `Settings` trait and bound `Bootstrap::Settings` by it. And both generic and edgehub settings have to implement the `Settings` trait. Which will be the same amount of code, imo.